federation/pkg/federation-controller/replicaset/replicasetcontroller.go

```diff
@@ -567,7 +570,7 @@ func (frsc *ReplicaSetController) reconcileReplicaSet(key string) (reconciliatio
 		lrs.Spec.Replicas = &specReplicas
 
 		if !exists {
-			if replicas > 0 {
+			if replicas > 0 && !isPuppet(frs) {
```

Comment:
I see this isPuppet sprinkled at many places in this function.
How about refactoring this function to create 2 smaller functions for:
* Updating cluster replicas spec as per federated replicaset; and
* Updating federated replicaset status as per cluster replicas.

This function (reconcileReplicaset) can call both these smaller functions.
We can skip calling the first one and call the second one for puppet replicasets. The 2 smaller functions dont need to know about puppet replicasets.